tfhe/src/shortint/ciphertext/glwe_packed_list.rs

```diff
@@ -0,0 +1,32 @@
+use crate::conformance::ParameterSetConformant;
+use crate::core_crypto::prelude::*;
+use crate::shortint::parameters::CiphertextConformanceParams;
+use crate::shortint::{CarryModulus, MessageModulus};
+
+use self::compressed_modulus_switched_glwe_ciphertext::CompressedModulusSwitchedGlweCiphertext;
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct GlwePackedCiphertextList {
+    pub compressed_modulus_switched_lwe_ciphertext:
+        Vec<CompressedModulusSwitchedGlweCiphertext<u64>>,
+    pub glwe_dimension: GlweDimension,
+    pub polynomial_size: PolynomialSize,
+    pub ciphertext_modulus: CiphertextModulus<u64>,
+    pub message_modulus: MessageModulus,
+    pub carry_modulus: CarryModulus,
+    pub pbs_order: PBSOrder,
+    pub lwe_per_glwe: usize,
+    pub count: usize,
+}
+
+impl ParameterSetConformant for GlwePackedCiphertextList {
+    type ParameterSet = CiphertextConformanceParams;
+
+    fn is_conformant(&self, params: &CiphertextConformanceParams) -> bool {
+        self.lwe_per_glwe <= self.polynomial_size.0
+            && self.ciphertext_modulus == params.ct_params.ct_modulus
+            && self.message_modulus == params.message_modulus
+            && self.carry_modulus == params.carry_modulus
+            && self.pbs_order == params.pbs_order
+    }
+}
```

Comment:
The idea was to check if a `GlwePackedCiphertextList` would be ok if successfully decompressed into a Ciphertext
But that does not seem very useful

I wonder if conformance is really useful compression as it happens server side (so if we trust the server, `GlwePackedCiphertextList` won't be broken)

Reply:
that's not the point of conformance, the point is to check that the deserialized data corresponds to the proper parameters to avoid panics later when using the list.

I'm trying to think about the blockchain case, but in their case there may be several parameter sets (like after parameter upgrade e.g.) so if the list is conformant for one set of parameter and not the other there is a risk of crashing

Reply:
If things are done properly server side, an valid serialized then deserialized object will still be valid

In my understanding, safe serialization is to avoid a untrusted user input to cause a late panic

Reply:
check the Glwe for conformance fully, this is not a proper conformance check

Reply:
Done